### PR TITLE
Grant kubernetes-release-test GCB write on k8s-staging-kubernetes GCR

### PR DESCRIPTION
**Pull request by justaugustus**
Grant the `kubernetes-release-test` Cloud Build account write access to
the container artifacts GCS location for `k8s-staging-kubernetes`
(`gs://artifacts.k8s-staging-kubernetes.appspot.com`).

This currently is a requirement as a domain restriction organization
policy is in place on the Google Infra projects, which prevents us from
completely moving staging to K8s Infra until dl.k8s.io is moved as well.

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

/assign @thockin @dims @cblecker 
cc: @tpepper @kubernetes/release-engineering 
ref: https://github.com/kubernetes/release/pull/1199, https://github.com/kubernetes/release/issues/270, https://github.com/kubernetes/release/issues/911, https://github.com/kubernetes/release/issues/1176

### CODE REVIEW

File: infra/gcp/ensure-staging-storage.sh [hunk 1]
```diff
@@ -80,6 +80,8 @@ RELEASE_STAGING_PROJECTS=(
     releng
 )
 
+RELEASE_OLD_STAGING_CLOUDBUILD_ACCOUNT="648026197307@cloudbuild.gserviceaccount.com"
```
Comment (listx) [inline infra/gcp/ensure-staging-storage.sh]: Nit: move "OLD" to the front

File: infra/gcp/ensure-staging-storage.sh [hunk 2]
```diff
@@ -190,6 +192,21 @@ for repo in "${RELEASE_STAGING_PROJECTS[@]}"; do
...
+    # Special case: Grant the 'kubernetes-release-test' Cloud Build account
+    # write access to the container artifacts GCS location for
+    # 'k8s-staging-kubernetes' ('gs://artifacts.k8s-staging-kubernetes.appspot.com').
+    #
+    # This currently is a requirement as a domain restriction organization
+    # policy is in place on the Google Infra projects, which prevents us from
+    # completely moving staging to K8s Infra until dl.k8s.io is moved as well.
+    #
+    # ref: https://github.com/kubernetes/release/issues/1176
+    if [[ $PROJECT == "k8s-staging-kubernetes" ]]; then
```
Comment (listx) [inline infra/gcp/ensure-staging-storage.sh]: Nit: `"${PROJECT}"` for consistency

File: infra/gcp/ensure-staging-storage.sh (hunk 1 above)
Comment (listx) [reply infra/gcp/ensure-staging-storage.sh]: Also, maybe comment this to say that this is the GCB build account for the `'kubernetes-release-test'` project, to line up with the comments in line 195.

**Comment by justaugustus** [comment]
@listx -- Fixed!

File: infra/gcp/ensure-staging-storage.sh [hunk 3]
```diff
@@ -190,6 +193,21 @@ for repo in "${RELEASE_STAGING_PROJECTS[@]}"; do
...
+    # Special case: Grant the 'kubernetes-release-test' Cloud Build service
+    # account write access to the container artifacts GCS location for
+    # 'k8s-staging-kubernetes' ('gs://artifacts.k8s-staging-kubernetes.appspot.com').
+    #
+    # This currently is a requirement as a domain restriction organization
+    # policy is in place on the Google Infra projects, which prevents us from
+    # completely moving staging to K8s Infra until dl.k8s.io is moved as well.
+    #
+    # ref: https://github.com/kubernetes/release/issues/1176
+    if [[ ${PROJECT} == "k8s-staging-kubernetes" ]]; then
```
Comment (listx) [inline infra/gcp/ensure-staging-storage.sh]: Micro-nit: `"${PROJECT}"` with quotes

/lgtm

Comment (justaugustus) [reply infra/gcp/ensure-staging-storage.sh]: @listx -- Fixed :)

**Comment by listx** [comment]
/lgtm

@thockin PTAL

**Comment by justaugustus** [comment]
Superseded by https://github.com/kubernetes/k8s.io/pull/675.
/close

**Pull request closed without merging**